manifests/internal/module/registerlayer.pp

```diff
@@ -14,6 +14,7 @@
     exec { "jboss::module::layer::${layer}":
       command => "/bin/awk -F'=' 'BEGIN {ins = 0} /^layers=/ { ins = ins + 1; print \$1=${layer},\$2 } END {if(ins == 0) print \"layers=${layer},base\"}' > ${jboss::home}/modules/layers.conf",
       unless  => "/bin/egrep -e '^layers=.*${layer}.*' ${jboss::home}/modules/layers.conf",
```

Comment:
If you are adding test for register layer, which is cool, you can also verify crucial command and unless parameters !resolve